app/uk/gov/hmrc/economiccrimelevyregistration/controllers/OtherEntityTypeController.scala

```diff
@@ -0,0 +1,97 @@
+
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.economiccrimelevyregistration.cleanup.OtherEntityTypeDataCleanup
+import uk.gov.hmrc.economiccrimelevyregistration.config.AppConfig
+import uk.gov.hmrc.economiccrimelevyregistration.connectors._
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.actions.{AuthorisedActionWithEnrolmentCheck, DataRetrievalAction}
+import uk.gov.hmrc.economiccrimelevyregistration.forms.OtherEntityTypeFormProvider
+import uk.gov.hmrc.economiccrimelevyregistration.forms.FormImplicits.FormOps
+import uk.gov.hmrc.economiccrimelevyregistration.handlers.ErrorHandler
+import uk.gov.hmrc.economiccrimelevyregistration.models._
+import uk.gov.hmrc.economiccrimelevyregistration.models.audit.OtherEntityTypeSelectedEvent
+import uk.gov.hmrc.economiccrimelevyregistration.navigation.OtherEntityTypePageNavigator
+import uk.gov.hmrc.economiccrimelevyregistration.views.html.OtherEntityTypeView
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class OtherEntityTypeController @Inject() (
+  val controllerComponents: MessagesControllerComponents,
+  authorise: AuthorisedActionWithEnrolmentCheck,
+  getRegistrationData: DataRetrievalAction,
+  eclRegistrationConnector: EclRegistrationConnector,
+  formProvider: OtherEntityTypeFormProvider,
+  pageNavigator: OtherEntityTypePageNavigator,
+  dataCleanup: OtherEntityTypeDataCleanup,
+  auditConnector: AuditConnector,
+  appConfig: AppConfig,
+  errorHandler: ErrorHandler,
+  view: OtherEntityTypeView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  val form: Form[OtherEntityType] = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (authorise andThen getRegistrationData) { implicit request =>
+    appConfig.privateBetaEnabled match {
+      case true => NotFound(errorHandler.notFoundTemplate(request))
+      case _    => Ok(view(form.prepare(request.registration.otherEntityJourneyData.entityType), mode))
+    }
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (authorise andThen getRegistrationData).async { implicit request =>
+    form
+      .bindFromRequest()
+      .fold(
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+        entityType => {
+          val otherEntityJourneyData = request.registration.otherEntityJourneyData.copy(
+            entityType = Some(entityType)
+          )
+
+//          auditConnector
+//            .sendExtendedEvent(
+//              OtherEntityTypeSelectedEvent(
+//                request.internalId,
+//                entityType
+//              ).extendedDataEvent
+//            )
```

Comment:
We need to remove this

Reply:
Leaving it acts as a  reminder that it needs to be sorted-out. We will need to audit this for sure.

Reply:
I don't want us to merge commented code like this to act as a reminder. The audit event needs to be agreed with CIP and there will be appropriate JIRA tickets etc. raised for us to do the work.

Reply:
Also - we may not even add an explicit event for this, as the 'Other' type is audited already, and the full data set is audited on submission

It requires further discussion either way

Reply:
Ok. Removed commented-out code.